salt/modules/pip.py

```diff
@@ -1301,6 +1305,73 @@ def list_(prefix=None, bin_env=None, user=None, cwd=None, env_vars=None, **kwarg
     return packages
 
 
+def list_(prefix=None, bin_env=None, user=None, cwd=None, env_vars=None, **kwargs):
+    """
+    .. versionchanged:: 3007.0
+
+    Output list of installed apps from ``pip list`` in JSON format and check to
+    see if ``prefix`` exists in the list of packages installed.
+
+    .. note::
+
+        If the version of pip available is older than 9.0.0, parsing the
+        ``freeze`` function output will be used to determine the name and
+        version of installed modules.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' pip.list salt
+    """
+
+    packages = {}
+    cwd = _pip_bin_env(cwd, bin_env)
+    cur_version = version(bin_env, cwd, user=user)
+
+    # Pip started supporting the ability to output json starting with 9.0.0
+    min_version = "9.0"
```

Comment:
I don't believe any of the platforms we currently support ships pip version 9.

Maybe we should just error out if it's that old?

Reply:
Any update if we can merge for 3006 as-is and then drop support for old versions in 3007?

Reply:
Per Open Hour conversation today, it's probably fine to go ahead and leave this for 3006. For 3007 we'll want to go ahead and drop support for pip 9 or less :+1: (should probably convert this comment to an issue)

Reply:
talked about this in Open Hour. deferring to 3007

Reply:
opened #63524